From e629e34f18141a6244eaac2f0288d7b2a5d669ee Mon Sep 17 00:00:00 2001
From: Chih-Hsuan Yen <yan12125@gmail.com>
Date: Sun, 3 Jul 2022 11:03:10 +0800
Subject: [PATCH] Fix the warning about missing modules.builtin.modinfo

> depmod: WARNING: could not open modules.builtin.modinfo at /tmp/mkinitcpio.EftMSK/root/lib/modules/5.15.52-1-lts: No such file or directory

kmod wants modules.builtin.modinfo since v30 [1].

Also adds modules.order as suggested by heftig. That file is used (if
exists) since the first commit of depmod.

[1] https://git.kernel.org/pub/scm/utils/kernel/kmod/kmod.git/commit/?id=0246e06340df292b5dda4bc00e24cc9ae894e881
---
 functions | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/functions b/functions
index 48cfd7a..e77a4c4 100644
--- a/functions
+++ b/functions
@@ -876,7 +876,7 @@ install_modules() {
     (( ${#zst_comp[*]} )) && zstd -d --rm -q "${zst_comp[@]}"
 
     msg "Generating module dependencies"
-    install -m644 -t "$moduledest" "$_d_kmoduledir"/modules.builtin
+    install -m644 -t "$moduledest" "$_d_kmoduledir"/modules.{builtin,builtin.modinfo,order}
 
     # we install all modules into kernel/, making the .order file incorrect for
     # the module tree. munge it, so that we have an accurate index. This avoids
